torch/distributed/_fsdp/fully_sharded_data_parallel.py

```diff
@@ -306,6 +296,41 @@ def __getitem__(self, key: int) -> Any:
         """Forward indexing calls in case the module is a nn.Sequential."""
         return self.module.__getitem__(key)  # type: ignore[operator]
 
+    def __getstate__(self) -> Dict[str, Any]:
+        """Serialize the state of the current FullyShardedDataParallel instance.
+        Some properties are not serializable (e.g., process groups, streams), so
+        we remove them and try to reconstruct them in :func:`__setstate__`.
+        """
+        state = copy.copy(self.__dict__)
+        state["is_sharded"] = [p._is_sharded for p in self.params]
+        state["orig_sizes"] = [p._orig_size for p in self.params]
+        if state["process_group"] is not None:
```

Comment:
if user attempts to pickle FSDP with non-default PG, and then unpickle it, it will use default process group and might create an issue with sharding? Should we throw an error when attempt to pickle with non-default process group?